packages/dashboard/src/components/dashboard/index.tsx

```diff
@@ -45,9 +49,14 @@ const Dashboard: React.FC<DashboardProperties> = ({
   edgeMode = 'disabled',
   initialViewMode,
   name,
+  currentViewport,
+  onViewportChange,
 }) => {
   useDashboardPlugins();
-  useDashboardViewport(dashboardConfiguration.defaultViewport);
+
+  const debounceOnViewportChange = onViewportChange
```

Comment:
debounced to prevent currentViewport from immediately changing every time a gesture happens